### PR TITLE
Add autocomplete and --list option for listing commands (#69)

### LINKED ISSUE #69: Autocompletion for shells
Auto completion for the cli options and available pages would be nice.
Something along the lines of the zsh and bash completions here: https://github.com/tldr-pages/tldr-node-client/tree/master/bin

This client currently does not have an option to list all available pages it seems, which would be a requirement.

### PR DESCRIPTION
**Pull request by wcheang**
Closes #69 

Hi! I was interested in the autocomplete feature, so I took a stab at implementing it. It didn't take much extra effort to add a --list option to list available commands for the platform, so I did that as well, seeing that the node client has one. I'm pretty new to contributing to opensource projects, so any suggestions are greatly appreciated :)

argcomplete was acting a little bit finicky on my local dev environment, so it'll be great if one of the maintainers can try it out to make sure it works fine as well. 

Thank you for this great client!

### CODE REVIEW

File: tldr.py [hunk 1]
```diff
@@ -221,6 +224,13 @@ def get_page(command, remote=None, platforms=None, languages=None):
 
 COMMAND_SPLIT_REGEX = re.compile(r'(?P<param>{{.+?}})')
 PARAM_REGEX = re.compile(r'(?:{{)(?P<param>.+?)(?:}})')
+CACHE_FILE_REGEX = re.compile(r'.*\/(.*)\.md')
+
+
+def get_commands(platforms=['**']):
+    for platform in platforms:
+        cache_files = glob(os.path.join(get_cache_dir(), 'pages', platform, '*.md'))
```
Comment (MasterOdin) [inline tldr.py:232]:
```suggestion
    cache_files = []
    for platform in platforms:
        cache_files += glob(os.path.join(get_cache_dir(), 'pages', platform, '*.md'))
```

File: tldr.py [hunk 2]
```diff
@@ -363,9 +378,11 @@ def main():
                         help='Override the default language')
 
     parser.add_argument(
-        'command', type=str, nargs='*', help="command to lookup"
+        'command', type=str, nargs='*', help="command to lookup", metavar='command',
+        choices=get_commands() + [[]]
```
Comment (MasterOdin) [inline tldr.py:382]: why the + `[[]]`?

**Comment by MasterOdin** [comment]
Thanks for the awesome patch and enabling autocompletion! It's one of the final big missing pieces for this client compared to the node one. Some minor critiques above, but the overall gist of it is spot-on!

File: tldr.py (hunk 2 above)
Comment (wcheang) [reply tldr.py:382]: When the user just runs `tldr` without any command, an empty list is passed in as the command. That leads to an error that says something like "[] is not a valid choice, choices: <the full list of command>" which looks horrible. This adds empty list to the list of choices and prevents the error.

File: tldr.py [hunk 3]
```diff
@@ -221,6 +224,13 @@ def get_page(command, remote=None, platforms=None, languages=None):
 
 COMMAND_SPLIT_REGEX = re.compile(r'(?P<param>{{.+?}})')
 PARAM_REGEX = re.compile(r'(?:{{)(?P<param>.+?)(?:}})')
+CACHE_FILE_REGEX = re.compile(r'.*\/(.*)\.md')
+
+
+def get_commands(platforms=['**']):
```
[the remark this answers is not in the document]
Comment (wcheang) [reply tldr.py:230]: Oh I didn't even see that there's a `get_platform_list()`, will update the code, thanks!

**Comment by wcheang** [comment]
I think all the comments should be addressed. @MasterOdin would you please have another look? Thanks!

**Comment by MasterOdin** [comment]
Thanks again for this.

**Pull request merged**